source/_static/clipboard-driver.js

```diff
@@ -0,0 +1,44 @@
+
+//Grab all elements with highlight-shell class - these will contain all shell commands
+var highlight_shell_elements = document.getElementsByClassName("highlight-shell");
+
+//Loop through elements with highlight-shell class and add clipboard button to each block
+for (x = 0; x < highlight_shell_elements.length; x++)
+{
+  //Create button element
+  var clipboard_button = document.createElement("BUTTON");
+
+  //Create span element and add glyphicon glyphicon-copy classes to span element
+  //Append span element as child of clipboard button
+  //Add clipboard-btn class to the clipboard button element
+  var span_element = document.createElement("SPAN");
+  span_element.setAttribute('class', 'glyphicon glyphicon-copy');
+  span_element.innerHTML = "";
+  clipboard_button.appendChild(span_element);
+  clipboard_button.setAttribute('class', 'clipboard-btn');
+
+  //Grab pre element
+  var pre_elements = highlight_shell_elements[x].getElementsByTagName("pre");
+
+  //Add the data-clipboard-text atribute to the clipboard button
+  //Assign the pre elements text content to the data-clipboard-text attribute
+  clipboard_button.setAttribute('data-clipboard-text', pre_elements[0].textContent);
+
+  //Style the clipboard button
+  //Display block
+  //Float right if it is not a "wget" or "curl" block
```

Comment:
I like where you're going with this, but something about this just feels a little strange here:

![screen shot 2018-03-30 at 2 34 06 pm](https://user-images.githubusercontent.com/274668/38154569-78771524-3427-11e8-8665-c0d5fd6bfbf6.png)

What if instead of putting the button inside the `pre` tag, we instead put it just above the code block:

![screen shot 2018-03-30 at 3 20 46 pm](https://user-images.githubusercontent.com/274668/38155703-f517e65c-342d-11e8-94c4-e761d43b175e.png)

This also has the advantage of removing the need [for the guarding](https://github.com/qiime2/docs/pull/294/files#diff-6f9b6d538886d4e4e60859eacd05b80cR32) based on block type below.

You can do that by going up a few levels with your selector:

```javascript
var commandBlockRoot = preElements[0].parentElement.parentElement;
commandBlockRoot.insertBefore(clipboardButton, commandBlockRoot.firstChild);
```